src/utils/rest-api.js

```diff
@@ -2407,9 +2408,17 @@ export function fetchMapEquipments(
 
     const substationParams = getQueryParamsList(substationsIds, 'substationId');
 
-    let fetchEquipmentsUrl =
-        getStudyUrlWithNodeUuid(studyUuid, currentNodeUuid) +
-        '/network-map/map-equipments';
+    let fetchEquipmentsUrl;
+    if (equipmentType === 'substations') {
+        fetchEquipmentsUrl =
+            getStudyUrlWithNodeUuid(studyUuid, currentNodeUuid) +
+            '/network-map/map-substations';
+    } else {
```

Comment:
It's not safer to check if the equipmentType is 'line' otherwise do a console.error ?

Reply:
I split it into 2 functions following @jonenst advice

Reply:
Ok

Reply:
Even if you split in 2 functions, I think both of these function should call the same function inside rest-api.js in fine. Because here all the code is duplicate and only one parameter change.